backend/contracts/ElectionFactory.sol

```diff
@@ -1,28 +1,51 @@
-//SPDX-License-Identifier: UNLICENSED
+// SPDX-License-Identifier: UNLICENSED
 
 pragma solidity ^0.8.0;
 
 contract ElectionFactory {
-    struct Election {
+    struct Candidate {
+        uint8 id;
         string name;
     }
 
-    Election[] public elections;
-
-    constructor() {
+    struct Ballot {
+        string name;
+        string description;
+        uint256 createTime;
+        uint256 votingTime;
+        Candidate[] candidates;
     }
 
-    function createElection(string memory _name) external{
-        Election memory newElection = Election(_name);
-        elections.push(newElection);
+    mapping(uint256 => Ballot) public ballots;
+    uint256 public nextBallotId;
+
+    event CandidateAdded(uint256 ballotId, uint8 id, string name);
+    event ContractInitialized(uint256 ballotId, string name, string description, uint256 createTime, uint256 votingTime, Candidate[] candidates);
+
+    function clearCandidates(uint256 _ballotId) internal {
+        delete ballots[_ballotId].candidates; // just a precaution for any random data that might have left in array
     }
 
-    function getAllElectionNames() external view returns (string[] memory) {
-        uint256 length = elections.length;
-        string[] memory names = new string[](length);
-        for (uint256 i = 0; i < length; i++) {
-            names[i] = elections[i].name;
+    function initialize(string memory _name, string memory _description, uint256 _votingTime, string[] memory _candidateNames) public {
```

Comment:
It's not clear what exactly this function does, when reading it's name. Consider renaming it into something like `createBallot`